### DISCUSSION TITLE
Discount based on Cart Value

### DISCUSSION THREAD

**Discussion by rdr6000** [medusajs/medusa#5440] [Q&A]
How can I implement discount or Free Shipping based on total cart value. Like "Free Shipping on orders above $15" ?

**Comment by bqst** [comment]
You can create a custom fulfillment service (see the doc: https://docs.medusajs.com/modules/carts-and-checkout/backend/add-fulfillment-provider) and override theses methods:

```
  async canCalculate(
    data: { [x: string]: unknown }
  ): Promise<boolean> {
    return true
  }

  // calculate price based on cart / order
  async calculatePrice(
    optionData: { [x: string]: unknown },
    data: { [x: string]: unknown },
    cart: Cart
  ): Promise<number> {
    const { subtotal } = cart
    let price = 0

    if (subtotal < 1500)
      price = 0
    } else {
      price = 500
    }
    return price
  }
```

Then, select your custom fulfillment service in Regions > Shipping Option > Fulfillment Method

**Reply by rdr6000** [reply]
Hey!! Sorry for the late reply. But it's not exactly what I was looking for. When we create a discount in admin ui we can set conditions. I am asking like can I set a discount condition where it activates if the cart value is above a certain amount. Like if cart value is above $15 15% off can be applied. 

We currently don't have any option to restrict a discount coupon to the cart value. It would be nice to have the option to do that since I am currently using a custom shipping provider from a 3rd party and it also does not encompass all the requirements as mentioned above. 

Any help will be appreciated. Thanks!!

**Reply by rdr6000** [reply]
Like here
![Screenshot_20231222-020758.png](https://github.com/medusajs/medusa/assets/40518544/a7edbd5c-d18c-4564-bd0d-e8682d0168e3)

Can I add a custom condition where the discount only activates based on Cart Value. It would be greatly appreciated for discounts such as "Shop above $15 and get 15% off, $10 and get 10% off, Free delivery above $20 cart value, etc."

Any help will be greatly appreciated. Thanks!!

**Reply by paramsinghvc** [reply]
Same question here! 
It's a shame such a common use case isn't addressed.

**Comment by rsoaresdev** [comment]
Hey! Recovering this issue, I want to enable free shipping when the costumer cart is +399$
How can I do it on MedusaJS 2.x?
It would be great if the was an option/condition on the Promotions section based on the cart price is higher, lower or equal something.
